spark/src/main/scala/org/apache/spark/sql/delta/CheckUnresolvedRelationTimeTravel.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.analysis.RelationTimeTravel
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.trees.TreePattern.RELATION_TIME_TRAVEL
+
+/**
+ * Custom check rule that compensates for [SPARK-45383]. It checks the (unresolved) child relation
+ * of each [[RelationTimeTravel]] in the plan, in order to trigger a helpful table-not-found
+ * [[AnalysisException]] instead of the internal spark error that would otherwise result.
+ */
+class CheckUnresolvedRelationTimeTravel(spark: SparkSession) extends (LogicalPlan => Unit) {
+  override def apply(plan: LogicalPlan): Unit = {
+    // Short circuit: We only care about (unresolved) plans containing [[RelationTimeTravel]].
+    if (plan.containsPattern(RELATION_TIME_TRAVEL)) {
```

Comment:
shall we add the pattern into delta time travel as well?
https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/catalyst/TimeTravel.scala

In this way, we need to test restore or clone with time travel.

Reply:
Delta's [PreprocessTimeTravel](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/PreprocessTimeTravel.scala#L81) rule already handles the `TimeTravel(UnresolvedRelation)` case, and so is not affected by the spark issue (`CheckAnalysis` should never encounter `TimeTravel` because we already threw if it couldn't be resolved).

This is a forward-looking fix -- eventually we want to eliminate `TimeTravel` entirely, and start using native spark `RelationTimeTravel` instead.

Reply:
Technically, the bug does impact Delta today -- if user intended to query a Delta table using spark's time travel syntax--but with a typo in the table name--then they would hit this bug and get confused by the unfriendly error message. The unit test validates this scenario. It's just a bit weird to talk about "Delta" here because the table doesn't exist in the first place.